Gen5Dashboard/Gen5Dashboard.cpp

```diff
@@ -12,10 +12,31 @@ Gen5Dashboard::Gen5Dashboard(int& argc, char** argv)
     , communicationContainer_(new CommunicationContainer())
     , businessContainer_(new BusinessContainer(*communicationContainer_, *dataContainer_))
     , presenterContainer_(new PresenterContainer(*dataContainer_))
-    , viewContainer_(new ViewContainer(*presenterContainer_))
 {
+    char mode = parseCommandLineArgs(argc, argv);
+    viewContainer_.reset(new ViewContainer(*presenterContainer_, mode));
```

Comment:
I don't think you'll need to reset here. Since this is the creation of the `viewContainer_`, just calling the constructor is fine.